src/main/java/org/reaktivity/nukleus/maven/plugin/internal/ast/parse/AstParser.java

```diff
@@ -585,8 +617,155 @@ private static int parseInt(
+    {
+        return Long.decode(text.substring(0, text.length() - 1));
+    }
+
+    private static String parseString(
+        RuleContext ctx)
+    {
+        return ctx != null ? ctx.getText() : null;
+    }
+
+    public final class EnumVisitor extends NukleusBaseVisitor<AstEnumNode.Builder>
+    {
+        private final AstEnumNode.Builder enumBuilder;
+        private AstValueNode.Builder valueBuilder;
+
+        public EnumVisitor()
+        {
+            this.enumBuilder = new AstEnumNode.Builder();
+        }
+
+        @Override
+        public AstEnumNode.Builder visitEnum_type(
+            Enum_typeContext ctx)
+        {
+            enumBuilder.name(ctx.ID().getText());
+
+            return super.visitEnum_type(ctx);
+        }
+
+        @Override
+        protected AstEnumNode.Builder defaultResult()
+        {
+            return enumBuilder;
+        }
+
+        @Override
+        public AstEnumNode.Builder visitInt8_type(
+            Int8_typeContext ctx)
+        {
+            enumBuilder.valueType(AstType.INT8);
+            return super.visitInt8_type(ctx);
+        }
+
+        @Override
+        public AstEnumNode.Builder visitInt16_type(
+            Int16_typeContext ctx)
+        {
+            enumBuilder.valueType(AstType.INT16);
+            return super.visitInt16_type(ctx);
+        }
+
+        @Override
+        public AstEnumNode.Builder visitInt32_type(
+            Int32_typeContext ctx)
+        {
+            enumBuilder.valueType(AstType.INT32);
+            return super.visitInt32_type(ctx);
+        }
+
+        @Override
+        public AstEnumNode.Builder visitInt64_type(
+            Int64_typeContext ctx)
+        {
+            enumBuilder.valueType(AstType.INT64);
+            return super.visitInt64_type(ctx);
+        }
+
+        @Override
+        public AstEnumNode.Builder visitString_type(
+            String_typeContext ctx)
+        {
+            enumBuilder.valueType(AstType.STRING);
+            return super.visitString_type(ctx);
+        }
+
+        @Override
+        public AstEnumNode.Builder visitString16_type(
+            String16_typeContext ctx)
+        {
+            enumBuilder.valueType(AstType.STRING16);
+            return super.visitString16_type(ctx);
+        }
+
+        @Override
+        public AstEnumNode.Builder visitString32_type(
+            String32_typeContext ctx)
+        {
+            enumBuilder.valueType(AstType.STRING32);
+            return super.visitString32_type(ctx);
+        }
+
+        @Override
+        public AstEnumNode.Builder visitEnum_value(
+            Enum_valueContext ctx)
+        {
+            this.valueBuilder = new AstValueNode.Builder()
+                .name(ctx.ID().getText())
+                .ordinal(enumBuilder.size());
+
+            AstEnumNode.Builder result = super.visitEnum_value(ctx);
+
+            AstValueNode value = valueBuilder.build();
+            enumBuilder.value(value);
+            return result;
```

Comment:
Suggest adding the following before the `return` to convey the intent of the code more clearly and prevent unintentional side effects in future.
```
this.valueBuilder = null;`
```